src/App.tsx

```diff
@@ -17,18 +37,32 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todosFromServer}
+                handleFilterChange={setFilteredTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodoId}
+                  handleSelectTodoId={setSelectedTodoId}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todo={selectedTodo}
+          onSetSelectedTodoId={setSelectedTodoId}
```

Comment:
Ensure that the `TodoModal` component uses the `onSetSelectedTodoId` function to clear or update the `selectedTodoId` when the modal is closed or the todo is modified. This will help maintain the correct state of the selected todo.